src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -49,12 +49,12 @@ public void save(
         claimantResponseRule.assertCanBeRequested(claim, claimantId);
 
         caseRepository.saveClaimantResponse(claim, response, authorization);
-        eventProducer.createClaimantResponseEvent(claim);
 
         if (response instanceof ResponseAcceptation && ((ResponseAcceptation) response).getFormaliseOption() != null) {
             formaliseResponseAcceptanceService.formalise(claim, (ResponseAcceptation) response, authorization);
         }
 
+        eventProducer.createClaimantResponseEvent(claim);
```

Comment:
Is there a test case where order of invocation using Mockito can be verified? Also does save needs to happen before formalise?

Reply:
Yes save will happen before formalise.